### ISSUE TITLE
[BUG] npm does not support Node.js vXX.XX.XX

### ISSUE THREAD

**Issue by jeme** [npm/cli#955]
# What / Why

I know posting this as a bug is a stretch, perhaps it's more a feature, but if felt like it was quite small for it to be that, improvement would be a better word.
----
Every now and then people come across that npm states something along the lines of:

```
1: npm WARN npm npm does not support Node.js v12.16.1
2: npm WARN npm You should probably upgrade to a newer version of node as we
3: npm WARN npm can't make any promises that npm will work with this version.
4: npm WARN npm Supported releases of Node.js are the latest release of 6, 8, 9, 10, 11.
5: npm WARN npm You can find the latest version at https://nodejs.org/
```

e.g. https://github.com/nodejs/help/issues/2134

This issue is not regarding that it happens, or the underlying cause of it, but rather that the warning provided by NPM is misleading and not at all to any help.

**2)** States that we should update to the latest version of node, however this likely happens as the user has just done just that, so it makes no sense in this context, and will just confuse the user.

**5)** Directes the user to nodejs.org to download the latest version, the very source from where he probably just did download and install node from (if on windows).

**1) and 4)** Will only confuse as the user is already on a version newer than what npm states it supports.

**3)** Out of scope, but this line might not be as relevant for this particular scenario, maybe if we are having a very old version of npm that hit's code that is removed from node.
----

Considering that npm logs the version of node as well as a list of supported versions, I would imagine that it wouldn't be too difficult to detect that Node is actually the one that is ahead, and the provide the appropriate help.

I you can point me to the right place to look I could have a look.

## When

When node is updated to a version that the current installed npm is unaware of, and npm fails to be updated along side of node.

### Current Behavior
NPM logs a warning that is of no help to the user in this scenario.
```
npm WARN npm npm does not support Node.js v12.16.1
npm WARN npm You should probably upgrade to a newer version of node as we
npm WARN npm can't make any promises that npm will work with this version.
npm WARN npm Supported releases of Node.js are the latest release of 6, 8, 9, 10, 11.
npm WARN npm You can find the latest version at https://nodejs.org/
```

### Steps to Reproduce
* n/a

### Expected Behavior
NPM logs a warning that is of help to the user in this scenario.

Helping the user to upgrade npm in whatever way that may be appropriate.

## Who
* n/a

## References
* n/a

**Comment by jeme** [comment]
This was fixed in NPM v6.14.0 by the looks of it, or rather the version check was fixed which makes the above obsolete.

Current version:
----
Range check based on: https://github.com/npm/cli/blob/latest/package.json#L312
Implemented at: https://github.com/npm/cli/blob/latest/lib/utils/unsupported.js#L11

NPM v6.13.7:
----
Implementation: https://github.com/npm/cli/blob/v6.13.7/lib/utils/unsupported.js#L20

**Comment by ljharb** [comment]
This should already be fixed in v6.14.0+
